### PR TITLE
boolbvt::print_assignment should call the print_assignment of the base class

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Comment by tautschnig** [comment]
Why?

**Review by romainbrenguier — APPROVED** [review]
This function does not seem to be used anywhere in the code base, so I guess it is used for debugging purposes. In that case, it may be worth adding that as a comment.

File: src/solvers/flattening/boolbv.cpp [hunk 1]
```diff
@@ -697,6 +697,8 @@ bool boolbvt::is_unbounded_array(const typet &type) const
 
 void boolbvt::print_assignment(std::ostream &out) const
 {
+  arrayst::print_assignment(out);
+
```
Comment (martin-cs) [inline src/solvers/flattening/boolbv.cpp:701]: I'd prefer SUB::print_assignment(out); and I do see @tautschnig 's point but otherwise OK.

**Review by martin-cs — APPROVED** [review]
Approve assuming changes will be made and questions answered.

**Comment by kroening** [comment]
Without the call to the base constructor, you simply don't get to see any variables in the lower layers (say the Boolean variables).

**Pull request merged**